platform/api/esx/api.go

```diff
@@ -282,6 +282,34 @@ func (a *API) GetConsoleOutput(name string) (string, error) {
 	return string(buf), nil
 }
 
+func (a *API) CleanupDevice(name string) error {
+	defaults, err := a.getServerDefaults()
+	if err != nil {
+		return fmt.Errorf("couldn't get server defaults: %v", err)
+	}
+
+	_, err = defaults.finder.VirtualMachine(a.ctx, name)
+	if err == nil {
+		return fmt.Errorf("VM still exists")
+	}
+
+	// Remove the serial.out file
+	fm := defaults.datastore.NewFileManager(defaults.datacenter, true)
+	uri := fmt.Sprintf("%s/serial.out", name)
+	err = fm.DeleteFile(a.ctx, uri)
+	if err != nil {
+		return fmt.Errorf("couldn't delete serial.out: %v", err)
```

Comment:
Ideally both of this and the next delete would return `nil` if the file didn't exist (because e.g. an administrator deleted it).

Not sure if the filemanager has a thing like `os.IsNotExist` to check for.